tests/integration/auth.spec.coffee

```diff
@@ -128,6 +128,51 @@ describe 'SDK authentication', ->
 
 				m.chai.expect(promise).to.be.rejectedWith('This email is already taken')
 
+		describe 'given an invalid token', ->
+
+			describe 'balena.auth.loginWithToken()', ->
+
+				it 'should be not rejected', ->
+					balena.auth.authenticate(credentials)
+					.then((token) -> balena.auth.loginWithToken("#{token}malformingsuffix"))
+					.then(balena.auth.getToken)
+					.then (key) ->
+						m.chai.expect(key).to.be.a('string')
+
+	describe 'when logged in with an invalid token', ->
+
+		before ->
+			balena.auth.logout()
+			.then ->
+				balena.auth.authenticate(credentials)
+			.then (token) ->
+				balena.auth.loginWithToken("#{token}malformingsuffix")
+
+		describe 'balena.auth.isLoggedIn()', ->
+
+			it 'should eventually be false', ->
+				promise = balena.auth.isLoggedIn()
+				m.chai.expect(promise).to.eventually.be.false
+
+		describe 'balena.auth.whoami()', ->
+
+			it 'should eventually be false', ->
```

Comment:
Quick typo: `it 'should eventually be undefined', ->`

Reply:
Nice catch @josecoelho :+1:

Reply:
@joehounsham here is a PR for this
See: https://github.com/balena-io/balena-sdk/pull/825